added regression ranking page, moved backcast functionality in the new page, updated template s/sheet

fidias-ieridis-steer opened the pull request:
added timestep on template s/sheets on cells H50 for now

fidias-ieridis-steer committed:
added regression ranking page, moved backcast functionality in that page, added timestep on template ssheets on cells H50 for now

Mark-Kerri commented on .gitignore:
```diff
@@ -166,5 +166,5 @@ cython_debug/
 # Data files
 *.csv
-*.xlsx
+#*.xlsx
```
Commenting this out is useful when you want to add a specific data file that is needed for the code to work. However, once you commit a version that uploads that spreadsheet to github, it's good to remove this hashtag so other data files are not inavertedly uploaded later. Once a file is uploaded once, it would keep updating regardless of the conditions in the gitignore.

fidias-ieridis-steer replied:
commit the specific spreadsheets, keeping xlsx in gitignore

Mark-Kerri commented on src/App.py:
```diff
@@ -105,13 +125,18 @@ def main():
         title="Regression Control",
```
I think we should delete the regression and backcast pages if we are now happy with the regression ranking page.

Mark-Kerri commented on src/apppages/read_inputs.py:
```diff
@@ -31,14 +31,16 @@ def main():
-        value=st.session_state.inputs_file_path,
-        # value=DEFAULT_FILE_PATH_FOR_TESTING, # use this when testing
+        # value=st.session_state.inputs_file_path,
+        value=DEFAULT_FILE_PATH_FOR_TESTING, # use this when testing
```
Minor thing but we should change this back to the broad input if pushing to main.

Mark-Kerri commented on data/reg_input/ Regression Inputs.xlsx:
I don't think we need this file on the github?

Mark-Kerri commented on src/apppages/regression_ranking.py:
```diff
@@ -0,0 +1,282 @@
+
+import streamlit as st
+import statsmodels.api as sm
+from apppages.utils.streamlit_tools import growth_df, stringify_g_df, create_and_show_df
```
This package list should be ordered by core party packages (e.g. itertools), third party packages (e.g. pandas), then our packages. Order below:

import itertools
import pandas as pd
import streamlit as st
import statsmodels.api as sm
import plotly.express as px
from apppages.utils.streamlit_tools import (
    stringify,
    growth_df,
    stringify_g_df,
    create_and_show_df,
)

Mark-Kerri commented on src/apppages/regression_ranking.py:
In general there are many occurances of lines being longer than 100 characters. I will push a version of this script to your branch which fixes it as I have the Black linter package. We should spend some time installing it on yours as it makes code as a whole much more readable.

Mark-Kerri replied:
I have also not reviewed the code logic in depth but testing it, it seems to work. 

One thing I would say is I think we can do the selection of a specific regression run better than copying the string name (although great for this stage), I think there is a way to add a button to each row for it to automatically run a command that brings the outputs below. 

Also, the backcast chart does not seem to change when I change the base year. You would expect the base year period to be exactly the same as the actual for the period selected then front/back cast from the ends of that period.

Mark-Kerri reviewed (commented):
I think we should address these comments before pushing to main, we could call to discuss first?

Mark-Kerri committed:
Formatting and gitignore

Edited gitignore to not upload excel files, and reformatted regression ranking.

fidias-ieridis-steer committed:
fixed bugs with regression_ranking

back-cast calculation fixed, added forecast calculation, now using base year slider properly

fidias-ieridis-steer committed:
Merge branch 'develop_regression_ranking' of https://github.com/Mark-Kerri/Traffic-Regression-Tool into develop_regression_ranking

# Conflicts:
#	src/apppages/regression_ranking.py

fidias-ieridis-steer committed:
added residuals outputs

fidias-ieridis-steer committed:
residuals chart, outputs page, export, slider bug fix

fidias-ieridis-steer committed:
input file without road names

fidias-ieridis-steer committed:
outputs page predicted y WIP

fidias-i committed:
v2 of output predict - WIP

fidias-ieridis-steer committed:
steer coloring on charts, chart export button (html)

fidias-ieridis-steer committed:
minor bug fixes and checks
